### PR TITLE
migration: log stages

### PR DESCRIPTION
**Pull request by Exca-DK**
Implements #984

### CODE REVIEW

File: migration/migration.go [hunk 1]
```diff
@@ -78,6 +79,9 @@ func MigrateIfNeeded(targetDB db.DB, network utils.Network) error {
 	}
 
 	for i := version; i < uint64(len(migrations)); i++ {
+		if log != nil {
+			log.Infow("applying database migration", "stage", fmt.Sprintf("%d/%d", i+1, uint64(len(migrations))))
```
Comment (omerfirmak) [inline migration/migration.go:83]:
```suggestion
			log.Infow("applying database migration", "stage", fmt.Sprintf("%d/%d", i+1, len(migrations)))
```

why do you need the casting here?

Comment (Exca-DK) [reply migration/migration.go:83]: Leftover from the previous commit, my bad.

Comment (joshklop) [inline migration/migration.go:83]:
```suggestion
			log.Infow("Applying database migration", "stage", fmt.Sprintf("%d/%d", i+1, uint64(len(migrations))))
```
Logs should be capitalized.

nit: we don't use the `if log != nil { ... }` and always assume the log is initialized. For consistency and to save a few lines here, I think we can safely use `log` directly. What do you think?

Comment (Exca-DK) [reply migration/migration.go:83]: Yeah, it's alright. I wasn't aware of that assumption.

**Pull request merged**